artemis-commons/src/main/java/org/apache/activemq/artemis/utils/ActiveMQThreadPoolExecutor.java

```diff
@@ -46,32 +45,70 @@
    private static class ThreadPoolQueue extends LinkedBlockingQueue<Runnable> {
 
       private ActiveMQThreadPoolExecutor executor = null;
+      private int idleThreads = 0;
 
       public void setExecutor(ActiveMQThreadPoolExecutor executor) {
          this.executor = executor;
       }
 
       @Override
-      public boolean offer(Runnable runnable) {
+      public synchronized boolean offer(Runnable runnable) {
```

Comment:
@bgutjahr Great, np to wait, after the changes I'll try to built a little benchmark on it to look if there will be any improvements/regressions that in a full broker test couldn't be seen :+1: 
This component is used pretty everywhere so I think it could worth to be sure of it